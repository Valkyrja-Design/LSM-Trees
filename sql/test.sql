create extension lsm;
create table lsmtest(id int, val int);
create table normal(id int, val int);
create index lsm_index on lsmtest using lsm(id);
create index idx on normal(id);

insert into lsmtest values (2,20);
select lsm_start_merge('lsm_index');
select lsm_wait_for_merge('lsm_index');
insert into lsmtest values (3,30);
select lsm_start_merge('lsm_index');
select lsm_wait_for_merge('lsm_index');
insert into lsmtest values (4,40);
-- select lsm_start_merge('lsm_index');
-- select lsm_wait_for_merge('lsm_index');
-- insert into lsmtest values (5,50);
-- select lsm_start_merge('lsm_index');
-- select lsm_wait_for_merge('lsm_index');
-- select lsm_get_merge_count('lsm_index');

select * from lsmtest where id = 1;
-- select * from lsmtest order by id;
-- select * from lsmtest order by id desc;
-- analyze lsmtest;
-- explain (COSTS OFF, TIMING OFF, SUMMARY OFF) select * from lsmtest order by id;

-- insert into lsmtest values (generate_series(1,100000), 1);
-- insert into lsmtest values (generate_series(1000001,200000), 2);
-- insert into lsmtest values (generate_series(2000001,300000), 3);
-- insert into lsmtest values (generate_series(1,100000), 1);
-- insert into lsmtest values (generate_series(1000001,200000), 2);
-- insert into lsmtest values (generate_series(2000001,300000), 3);
-- select * from lsmtest where id = 1;
-- select * from lsmtest where id = 1000000;
-- select * from lsmtest where id = 2000000;
-- select * from lsmtest where id = 3000000;
-- analyze lsmtest;
-- explain (COSTS OFF, TIMING OFF, SUMMARY OFF) select * from lsmtest where id = 1;
-- select lsm_get_merge_count('lsm_index') > 5;

-- truncate table lsmtest;
-- insert into lsmtest values (generate_series(1,1000000), 1);
-- select * from lsmtest where id = 1;

-- reindex table lsmtest;
-- select * from lsmtest where id = 1;

