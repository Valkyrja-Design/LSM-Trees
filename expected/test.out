CREATE EXTENSION
CREATE TABLE
CREATE TABLE
CREATE INDEX
CREATE INDEX
INSERT 0 1
 lsm_start_merge 
-----------------
 
(1 row)

 lsm_wait_for_merge 
--------------------
 
(1 row)

INSERT 0 1
 lsm_start_merge 
-----------------
 
(1 row)

 lsm_wait_for_merge 
--------------------
 
(1 row)

INSERT 0 1
 lsm_start_merge 
-----------------
 
(1 row)

 lsm_wait_for_merge 
--------------------
 
(1 row)

INSERT 0 1
 lsm_start_merge 
-----------------
 
(1 row)

 lsm_wait_for_merge 
--------------------
 
(1 row)

 lsm_get_merge_count 
---------------------
                   4
(1 row)

 id | val 
----+-----
(0 rows)

 id | val 
----+-----
  2 |  20
  3 |  30
  4 |  40
  5 |  50
(4 rows)

 id | val 
----+-----
  5 |  50
  4 |  40
  3 |  30
  2 |  20
(4 rows)

ANALYZE
        QUERY PLAN         
---------------------------
 Sort
   Sort Key: id
   ->  Seq Scan on lsmtest
(3 rows)

INSERT 0 100000
INSERT 0 0
INSERT 0 0
INSERT 0 100000
INSERT 0 0
INSERT 0 0
 id | val 
----+-----
  1 |   1
  1 |   1
(2 rows)

 id | val 
----+-----
(0 rows)

 id | val 
----+-----
(0 rows)

 id | val 
----+-----
(0 rows)

ANALYZE
              QUERY PLAN               
---------------------------------------
 Index Scan using lsm_index on lsmtest
   Index Cond: (id = 1)
(2 rows)

 ?column? 
----------
 f
(1 row)

TRUNCATE TABLE
INSERT 0 1000000
 id | val 
----+-----
  1 |   1
(1 row)

REINDEX
 id | val 
----+-----
  1 |   1
(1 row)
